oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtGrantedAuthoritiesConverter.java

```diff
@@ -117,16 +122,38 @@ private String getAuthoritiesClaimName(Jwt jwt) {
 		return null;
 	}
 
+	/**
+	 * Sets the expression for extracting the token claim to use for mapping {@link GrantedAuthority
+	 * authorities} by this converter.
+	 * Note that this takes precedence over {@link #setAuthoritiesClaimName(String)}.
+	 * @param authoritiesClaimExpression The token claim SpEL Expression to map authorities
+	 * @since 6.5
+	 */
+	public void setAuthoritiesClaimExpression(Expression authoritiesClaimExpression) {
+		Assert.notNull(authoritiesClaimExpression, "authoritiesClaimExpression must not be null");
+		this.authoritiesClaimExpression = authoritiesClaimExpression;
+	}
+
 	private Collection<String> getAuthorities(Jwt jwt) {
-		String claimName = getAuthoritiesClaimName(jwt);
-		if (claimName == null) {
-			this.logger.trace("Returning no authorities since could not find any claims that might contain scopes");
-			return Collections.emptyList();
-		}
-		if (this.logger.isTraceEnabled()) {
-			this.logger.trace(LogMessage.format("Looking for scopes in claim %s", claimName));
+
+		Object authorities;
+		if (authoritiesClaimExpression != null) {
```

Comment:
Given that this is a separate strategy, perhaps this would be better in a separate class. I think something like `ExpressionJwtGrantedAuthoritiesConverter` makes sense. Would you mind changing to that instead?

Reply:
Thanks for the review! I followed your suggestion and introduced `ExpressionJwtGrantedAuthoritiesConverter`. 
I think this could be used in combination with `DelegatingJwtGrantedAuthoritiesConverter` to combine `GrantedAuthority`'s from `JwtGrantedAuthoritiesConverter`.